src/features/auth/components/NoAccount.tsx

```diff
@@ -14,50 +17,114 @@ import { useActor } from "@xstate/react";
 import { SUNNYSIDE } from "assets/sunnyside";
 import { useAppTranslation } from "lib/i18n/useAppTranslations";
 import { Loading } from "./Loading";
+import {
+  checkReferralCode,
+  getReferrerId,
+  saveReferrerId,
+} from "../actions/createAccount";
+import { ConfirmationModal } from "components/ui/ConfirmationModal";
+import { TextInput } from "components/ui/TextInput";
+import debounce from "lodash.debounce";
 
 export const NoAccount: React.FC = () => {
+  const { t } = useAppTranslation();
   const { authService } = useContext(Context);
   const [authState] = useActor(authService);
 
-  const [showPromoCode, setShowPromoCode] = useState(false);
-  const [promoCode, setPromoCode] = useState(getPromoCode());
-
+  const [showReferralId, setShowReferralId] = useState(false);
+  const [referralId, setReferralId] = useState(getReferrerId() ?? undefined);
+  const [showEmptyReferralModal, setShowEmptyReferralModal] = useState(false);
+  const [validationState, setValidationState] = useState<
```

Comment:
Will these work in different languages? I would suggest using the type `TranslationKey` and parsing it into `t()` when you display to users.